pkg/tfshim/shim.go

```diff
@@ -343,3 +351,22 @@ type DiffOptions struct {
 //
 // https://www.pulumi.com/docs/concepts/options/ignorechanges/
 type IgnoreChanges = func() map[string]struct{}
+
+// RawState is the raw un-encoded Terraform state, without type information. It is passed as-is for providers to
+// upgrade and run migrations on.
+//
+// The representation is the result of parsing the JSON format accepted on the gRPC Terraform protocol:
+//
+//	https://github.com/hashicorp/terraform-plugin-go/blob/v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5.pb.go#L519
+//	https://github.com/hashicorp/terraform-plugin-go/blob/v0.26.0/tfprotov6/state.go#L35
+type RawState any
```

Comment:
Go is making me do typing mistakes here as it automatically accepts anything as RawState... I might refactor this to hide the implementation behind a struct or such so that conversion is always explicit.